clients/extension/src/background/handlers/accountsHandler.ts

```
@@ -28,17 +31,48 @@ export const handleFindAccounts = async (
 
   if (!currentVaultId) return []
   const vaultSessions = await getVaultAppSessions(currentVaultId)
-
   const currentSession = vaultSessions[getDappHostname(sender)] ?? null
 
-  if (currentSession) {
-    const vaultsCoins = await getVaultsCoins()
-    return vaultsCoins[currentVaultId]
-      .filter(
-        accountCoin => isFeeCoin(accountCoin) && accountCoin.chain === chain
-      )
-      .map(({ address }) => address ?? '')
-  } else {
+  // First check if we have the address in our coins
+  const vaultsCoins = await getVaultsCoins()
+  const existingCoins = vaultsCoins[currentVaultId]
+    .filter(
+      accountCoin => isFeeCoin(accountCoin) && accountCoin.chain === chain
+    )
+    .map(({ address }) => address ?? '')
```

Let's avoid using this kind of notation `?? ''`, as it indicates that we are intentionally setting the value to an invalid one. If something is wrong we should throw an error, I don't see why there will be a missing address.

---

If you need to find a single account, why use filter and map instead of using find and naming it existingCoin? Why should there be multiple coins for a chain? Also why call it coins, if those are addresses?

---

Maybe this function should be renamed if there could be only a single account?

---

I renamed existingCoins to use more accurate terminology (account instead of coin) and simplified the existing logic by directly returning the address if it's found, to match the new changes.